src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -1421,13 +1446,8 @@ public ICollection<TValue> Values
         /// found and removed; otherwise, false.</returns>
         /// <exception cref="System.ArgumentNullException">The Key property of <paramref
         /// name="keyValuePair"/> is a null reference (Nothing in Visual Basic).</exception>
-        bool ICollection<KeyValuePair<TKey, TValue>>.Remove(KeyValuePair<TKey, TValue> keyValuePair)
-        {
-            if (keyValuePair.Key == null) throw new ArgumentNullException(nameof(keyValuePair), SR.ConcurrentDictionary_ItemKeyIsNull);
-
-            TValue throwAwayValue;
-            return TryRemoveInternal(keyValuePair.Key, out throwAwayValue, true, keyValuePair.Value);
-        }
+        bool ICollection<KeyValuePair<TKey, TValue>>.Remove(KeyValuePair<TKey, TValue> keyValuePair) =>
+            TryRemove(keyValuePair);
```

Comment:
This changes the name of the argument captured in the `ArgumentNullException`. TBH I don't see this as a problem, but just wanted to confirm that this was intentional.

Reply:
> just wanted to confirm that this was intentional

It is intentional.  The name of the parameter on the interface is `item`; I have no idea where this `keyValuePair` came from, but it's only going to be visible to someone via reflection... anyone calling this API via the interface (which is the only way you can call it, since it's explicitly defined) will see `item` in the signature, in the docs, etc... even the message in the exception says "item.Key": `TKey is a reference type and item.Key is null.`.